src/nti/externalization/_datastructures.pxd

```diff
@@ -42,7 +42,7 @@ cdef class ExternalizableDictionaryMixin(object):
 
 cdef class AbstractDynamicObjectIO(ExternalizableDictionaryMixin):
     cpdef _ext_replacement(self)
-    cpdef _ext_all_possible_keys(self)
+    cpdef frozenset _ext_all_possible_keys(self)
```

Comment:
@papachoco  I see (now) that `nti.namedfile` defines `class NamedFileObjectIO(AbstractDynamicObjectIO):` which returns `()` from `_ext_all_possible_keys()`; that will raise `TypeError` with this change until it is updated to return `frozenset()`.

Reply:
I should have that [fixed here](https://github.com/NextThought/nti.namedfile/pull/1)

Reply:
Thanks. I didn't think about it at the time, I had just noticed that cython generates specific (faster) code for frozenset iteration vs the generic code it generates otherwise, saw that we were taking advantage of that in other places and just dropped it in here.